Enterprise-CMCS/macpro-mdct-mfp
# Make completion status work for dynamicModalOverlays

### Description

Make the completion status show that green checkmark thingy that gives the happy feels instead of the red exclaimation thingy that makes you sad

https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/19439679/34715f60-04de-4056-8d0f-5705c6b50b8e

Things I'd like to do at a later time

- Fix up how completionStatus matches fieldData to form templates for initiatives. Too many for loops. Way too complex.
- Validation for choicelists in the initiatives isn't showing validation correctly
- Breakup evaluationPlan and fundingSources extraction to separate functions for cleaner tests
- Write more tests for the different objective card scenarios (what data has to show and when)
- Remove the unused data we don't need from workplan thats being pass to the SAR
- Confirm whether completionStatus needs to see at least 4 types of initiatives in order to submit or not

### Related ticket(s)


CMDCT-3175

---
### How to test

Create a SAR
Check review and submit page to see its not completed
Fill out the State and Territory Objectives page
See the green checkmark appear!

---
### Author checklist
